alacritty/src/input.rs

```diff
@@ -455,89 +455,78 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
         }
     }
 
-    fn on_mouse_double_click(&mut self, button: MouseButton, point: Point) {
-        if button == MouseButton::Left {
-            let side = self.ctx.mouse().cell_side;
-            self.ctx.start_selection(SelectionType::Semantic, point, side);
+    fn on_mouse_press(&mut self, button: MouseButton) {
+        // Handle mouse mode
+        if !self.ctx.modifiers().shift() && self.ctx.mouse_mode() {
+            self.ctx.mouse_mut().click_state = ClickState::None;
+
+            let code = match button {
+                MouseButton::Left => 0,
+                MouseButton::Middle => 1,
+                MouseButton::Right => 2,
+                // Can't properly report more than three buttons.
+                MouseButton::Other(_) => return,
+            };
+
+            self.mouse_report(code, ElementState::Pressed);
+            return;
```

Comment:
Probably the biggest thing standing out in the current state of this PR is these two early returns.

While the `button != Left` early return is fairly small and straight-forward, this is some pretty big chunk of code for an early return. That's usually a bit of a code smell and increases the complexity of understanding the code a bit.

The method is also already 73 lines long, which is relatively big I'd say and is quickly approaching the limits of what I'd say is acceptable.

That's why I would suggest moving everything after these early returns to a separate function for handling just LMB clicks when mouse mode is not enabled.

That would also allow getting rid of early returns by just doing `if { ... } else if button != Left { ... } else { lmb_stuff() }`.